solutions/usaco-1061.mdx

```diff
@@ -0,0 +1,109 @@
+---
+id: usaco-1061
+source: USACO Bronze 2020 December
+title: Stuck in a Rut
+author: Ryan Chou
+---
+
+[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_prob3_bronze_dec20.html)
+
+## Solution
+
+Since $x, y \leq 10^9$, we can't simulate each time period. Instead, we should look for signs which allow two cows to intersect. 
+
+One important observation is that cows can't move backwards, so the only way for two cows to collide is if $n[x] > e[x] $ and $n[y] < e[y]$. 
+
+This ensures that the cows are positioned like this:
+
+![](USACO-dec20-stuckinarut-bronze/intersection_example.png)
+
+From here, we can track which cows will collide and how much grass they'll eat.
```

Comment:
This seems a bit too vague.

I'd elaborate more on
1) Sorting (so the first stoppings occur first)
2) An intersection that would have occurred may fail to occur because a cow is stopped early. Sorting resolves this because...

Reply:
👍

Reply:
> An intersection that would have occurred may fail to occur because a cow is stopped early. Sorting resolves this because...
We still have to check if any of the cows were stopped before that, since we're scanning every pair of cows and checking if they collide.